### ISSUE TITLE
Fixing DocStart redirect loops

### ISSUE THREAD

**Issue by bovine3dom** [cmcaine/tridactyl#516]
Sometimes, Amazon decides to redirect you from smile.amazon.co.uk to www.amazon.co.uk if you aren't logged in.

Then, `autocmd DocStart www.amazon.co.uk urlmodify -t www smile` sends you into a loop / DOS.

Perhaps we should have an option to make DocStart only run if the page wasn't accessed via a redirect? Or some kind of rate limiter for DocStart?

What do other people think?

**Comment by glacambre** [comment]
We could also have DocStart grab a per-tab lock before doing anything and then release it when the document has loaded.

**Comment by rddunphy** [comment]
Just ran into this issue. Only running DocStart if the page wasn't accessed via a redirect would probably prevent a lot of unproblematic redirects, not just loops. Would it make more sense to avoid performing the same redirect twice within a short period?

**Comment by bovine3dom** [comment]
Yeah, maybe we could add a "safe" mode to `urlmodify` that would only let you call it e.g. twice a second.
